BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m

```diff
@@ -742,7 +756,7 @@ - (void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTas
             id<BOXURLSessionTaskDelegate> taskDelegate = [self taskDelegateForSessionId:session.configuration.identifier sessionTaskId:downloadTask.taskIdentifier];
             NSString *destinationFilePath = nil;
 
-            if ([taskDelegate conformsToProtocol:@protocol(BOXURLSessionDownloadTaskDelegate)]) {
+            if ([taskDelegate conformsToProtocol:@protocol(BOXURLSessionDownloadTaskDelegate)] && [taskDelegate respondsToSelector:@selector(destinationFilePath)]) {
```

Comment:
If we have respondToSelector check, we probably don't need conformsToProtocol. It'll have to conform to protocol to respond to the selector.